event_sourcery_esdb/event_store.py

```diff
@@ -94,3 +102,44 @@ def _get_stream_position(self, name: stream.Name) -> stream.Position | None:
     def delete_stream(self, stream_id: StreamId) -> None:
         name = stream.Name(stream_id)
         self._client.delete_stream(str(name), current_version=StreamState.ANY)
+
+    def subscribe(
+        self,
+        from_position: Position | None,
+        to_category: str | None,
+        to_events: list[str] | None,
+    ) -> Iterator[RecordedRaw]:
+        if from_position is None:
+            from_position = self.current_position
+        if to_category is not None:
+            subscription = self._client.subscribe_to_all(
+                commit_position=from_position,
+                timeout=self._timeout,
+                filter_include=[
+                    f"{to_category}-\\w+",
+                ],
+                filter_by_stream_name=True,
+            )
+        elif to_events is not None:
+            subscription = self._client.subscribe_to_all(
+                commit_position=from_position,
+                timeout=self._timeout,
+                filter_include=to_events,
+                filter_by_stream_name=False,
+            )
+        else:
+            subscription = self._client.subscribe_to_all(
+                commit_position=from_position,
+                timeout=self._timeout,
+            )
+        return (
+            RecordedRaw(
+                entry=dto.raw_event(from_entry=recorded),
+                position=Position(recorded.commit_position or 0),
+            )
+            for recorded in subscription
+        )
+
+    @property
+    def current_position(self) -> Position | None:
```

Comment:
I must say I don't get the full picture. This property is not really used outside of the class, right? Meaning it could be private?

Reply:
It's added to the protocol of the storage strategy.

Reply:
It's exposed later in EventStore.position